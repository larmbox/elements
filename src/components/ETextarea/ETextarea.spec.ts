import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ETextarea } from '~/index';

test('mount component', () => {
  expect(ETextarea).toBeTruthy();

  const component = mount(ETextarea, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('ETextarea');
  expect(component.html()).toContain('EFormControl--field');
});

test('register css variables', () => {
  mount(ETextarea);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('ETextarea');
});

test('inherit attributes', () => {
  const component = mount(ETextarea, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
