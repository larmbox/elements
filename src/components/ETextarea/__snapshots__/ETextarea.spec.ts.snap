// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl EFormControl--field ETextarea ETextarea--md ETextarea--primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"ETextarea-inner\\"><textarea id=\\"id\\" style=\\"resize: both;\\"></textarea></div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
